controllers/ipc_controller.go

```diff
@@ -177,6 +183,62 @@ func (r *IPConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (r
 	}
 }
 
+// gateIPConfigByIBU gates IPConfig reconciliation based on the ImageBasedUpgrade (IBU) CR.
+//
+// IPC is considered "allowed to proceed" only when:
+//   - the IBU CR exists, AND
+//   - IBU spec.stage is Idle, AND
+//   - the IBU Idle condition is present and true.
+//
+// Otherwise IPC gates reconciliation as follows:
+//   - If the IBU CR does not exist yet: requeue immediately (without updating IPC status).
+//   - If the IBU CR exists but has no status conditions yet: mark IPC as Blocked and requeue immediately.
+//   - If IBU is not idle (stage != Idle or Idle condition is not true): mark IPC as Blocked and requeue after a short interval.
+//
+// Notice: this is not mutual exclusion. For delivery reasons, IBU can still start while IPC is
+// not Idle. We should add mutual exclusion between IPC and IBU controllers to avoid this
+// in the future.
+func (r *IPConfigReconciler) gateIPConfigByIBU(
+	ctx context.Context,
+	ipc *ipcv1.IPConfig,
+) (ctrl.Result, error) {
+	ibu := &ibuv1.ImageBasedUpgrade{}
+	if getErr := r.NoncachedClient.Get(ctx, client.ObjectKey{Name: controllerutils.IBUName}, ibu); getErr != nil {
+		if !k8serrors.IsNotFound(getErr) {
+			return requeueWithError(fmt.Errorf("failed to get ImageBasedUpgrade for gating: %w", getErr))
+		}
+		return requeueImmediately(), nil
+	}
+
+	if len(ibu.Status.Conditions) == 0 {
+		controllerutils.SetIPStatusBlocked(ipc, "Blocked by gating: IBU is not initialized")
+		if err := r.Client.Status().Update(ctx, ipc); err != nil {
+			return requeueWithError(fmt.Errorf("failed to update ipconfig status: %w", err))
+		}
+		return requeueImmediately(), nil
+	}
+
+	if ibu.Spec.Stage == ibuv1.Stages.Idle &&
```

Comment:
Add some logs here, e.g.
IBU is idle + IBU idle status condition
Removing the blocked condition